examples/sampling/pyg/test.py

```diff
@@ -0,0 +1,191 @@
+import dgl.graphbolt as gb
+import torch
+import torch.nn.functional as F
+from torch_geometric.data import Data
+from torch_geometric.loader import NeighborSampler
+from torch_geometric.nn import SAGEConv
+
+
+class GraphSAGE(torch.nn.Module):
+    def __init__(self, in_channels, hidden_channels, out_channels):
+        super(GraphSAGE, self).__init__()
+        self.conv1 = SAGEConv(in_channels, hidden_channels)
+        self.conv2 = SAGEConv(hidden_channels, out_channels)
+
+    def forward(self, x, adjs):
+        for i, (edge_index, _, size) in enumerate(adjs):
+            x_target = x[: size[1]]
+            if i == 0:
+                x = self.conv1((x, x_target), edge_index)
+                x = F.relu(x)
+            else:
+                x = self.conv2((x, x_target), edge_index)
+                x = F.relu(x)
+            if i != len(adjs) - 1:
+                x = F.dropout(x, p=0.5, training=self.training)
+        return x
+
+
+def load_data():
+    dataset = gb.BuiltinDataset("ogbn-arxiv").load()
+    graph = dataset.graph
+    features = dataset.feature
+    features_tensor = features.read("node", None, "feat")
+    train_set = dataset.tasks[0].train_set
+    valid_set = dataset.tasks[0].validation_set
+    test_set = dataset.tasks[0].test_set
+    train_labels = train_set._items[1]
+    valid_labels = valid_set._items[1]
+    test_labels = test_set._items[1]
+    graph = dataset.graph
+    csc_indptr = graph.csc_indptr
+    rows = torch.repeat_interleave(
+        torch.arange(csc_indptr.size(0) - 1, device=csc_indptr.device),
+        csc_indptr[1:] - csc_indptr[:-1],
+    )
+    indices = graph.indices
+    cols = indices
+    edge_index = torch.stack([rows, cols], dim=0)
+    pyg_data = Data(x=features_tensor, edge_index=edge_index, y=train_labels)
+    num_nodes = graph.total_num_nodes
+
+    return (
+        pyg_data,
+        train_labels,
+        valid_labels,
+        test_labels,
+        train_set,
+        valid_set,
+        test_set,
+        num_nodes,
+    )
+
+
+def create_data_loaders(
+    pyg_data,
+    train_labels,
+    valid_labels,
+    test_labels,
+    train_set,
+    valid_set,
+    test_set,
+):
+    train_mask = train_set._items[0]
+    valid_mask = valid_set._items[0]
+    test_mask = test_set._items[0]
+    train_loader = NeighborSampler(
```

Comment:
It looks like the 3 different dataloader creation portions can be refactored, looks like only the `node_idx` and `shuffle` parameters are different. Should make the example more concise.

Reply:
yessir